### ISSUE TITLE
ERROR: KeyError: key 60096 not found

### ISSUE THREAD

**Issue by liamh** [sisl/SatelliteDynamics.jl#18]
This appears to be coming from an attempted  MSIS data download.
``` julia
using SatelliteDynamics
pv = [1.0223996343275267e6, -6.548808273197852e6, 0.0, 4036.3406898714115, 630.1533154108832, 6617.161182506771]
orb  = EarthInertialState(Epoch(2023, 6, 1, 12, 0, 0, 0.0), pv, dt=1.0,
                          mass=100.0, coef_drag=1.0, area_drag=1.0, #B=0.01
                          n_grav=0, m_grav=0,
                          drag=true, srp=false,
                          moon=false, sun=false,
                          relativity=false)
t, epc, eci = sim!(orb, 86400.0)

ERROR: KeyError: key 60096 not found
Stacktrace:
  [1] getindex
    @ ./dict.jl:484 [inlined]
  [2] ApDailyIndex
    @ ~/.julia/packages/SatelliteDynamics/JGriC/src/earth_environment/space_weather.jl:183 [inlined]
  [3] ApDailyIndex(mjd::Float64)
    @ SatelliteDynamics ~/.julia/packages/SatelliteDynamics/JGriC/src/earth_environment/space_weather.jl:186
  [4] density_nrlmsise00(epc::Epoch, x::Vector{Float64}; use_degrees::Bool)
    @ SatelliteDynamics ~/.julia/packages/SatelliteDynamics/JGriC/src/earth_environment/nrlmsise00.jl:1630
  [5] density_nrlmsise00
    @ ~/.julia/packages/SatelliteDynamics/JGriC/src/earth_environment/nrlmsise00.jl:1608 [inlined]
  [6] fderiv_earth_orbit(epc::Epoch, x::Vector{Float64}; mass::Float64, area_drag::Float64, coef_drag::Float64, area_srp::Float64, coef_srp::Float64, n_grav::Int64, m_grav::Int64, drag::Bool, srp::Bool, moon::Bool, sun::Bool, relativity::Bool)
    @ SatelliteDynamics ~/.julia/packages/SatelliteDynamics/JGriC/src/simulation/propagators.jl:60
  [7] istep(rk4::RK4, epc::Epoch, dt::Float64, x::Vector{Float64})
    @ SatelliteDynamics ~/.julia/packages/SatelliteDynamics/JGriC/src/simulation/integrators.jl:80
  [8] step!(state::EarthInertialState, dt::Float64)
    @ SatelliteDynamics ~/.julia/packages/SatelliteDynamics/JGriC/src/simulation/propagators.jl:157
  [9] sim!(state::EarthInertialState, time::Float64)
    @ SatelliteDynamics ~/.julia/packages/SatelliteDynamics/JGriC/src/simulation/propagators.jl:255
 [10] top-level scope
    @ REPL[4]:1
```

**Comment by duncaneddy** [comment]
So this issue is caused by the fact that data currently bundled with the package doesn't go through the requested date (2023-06-01, MJD 60096), the last bundled data is from 2023-04-13.

Normally you could address this by calling `download_all_data()` from the REPL once to update the package data files, but it looks like the source server changed their hosting location so I'll need to go in, find a new source for that data, write a new parser, and update the data if you need to use the propagation on that exact date. That is going to take me a bit to get to, though I'm happy to accept a PR as well for it.

An immediate solution would be to just choose an earlier date for the propagation, something like 2023-01-01.

**Comment by liamh** [comment]
It would helpful to have a more explanatory error message that says basically what you said in your comment. However, I was actually hoping to be able to use Harris-Priester instead of MSIS, and I noticed that the H-P model is in this package, but there seems to be no way to select it in place of MSIS. I wrote a one-line fix to call H-P, but it doesn't work for me even manually loading the one calling function I patched, it always uses this repository version. (I am completely new to Julia so I don't know how to patch a package.)

So my suggestion has two parts: make this error message more meaningful to the average user (something like "date requested 2023-06-01 is beyond the end of the available MSIS data (2023-04-13), choose an earlier date"), and make it possible to switch atmosphere models.

**Comment by duncaneddy** [comment]
Thank you for posting the fix and the current data sources for both files, as well as reminding me about this issue.

I plan to go in and clean up the code to:
1. Fix `download_all_data()` so it properly updates data sources
1. Download the latest data sources for all files to get the back up to date
1. Migrate package CI GitHub Actions
1. Setup an Action to perform a daily package data update
1. Write a more helpful error message on out-of-bounds access to inform the user to update data.

**Comment by duncaneddy** [comment]
All of these issues should now be addressed with the `v0.5.0` release of the package. I have completed implementing all fixes and improvements outlined above, so hopefully these issues will be more understandable in the future.
